google-cloud-tools-plugin/testSrc/com/google/cloud/tools/intellij/apis/CloudLibraryDependencyWriterTest.java

```diff
@@ -0,0 +1,159 @@
+
+              MavenDomDependency dependency = cloudLibraryDependencies.get(0);
+              assertThat(dependency.getGroupId().getStringValue()).isEqualTo(CLOUD_LIB_GROUP_ID);
+              assertThat(dependency.getArtifactId().getStringValue())
+                  .isEqualTo(CLOUD_LIB_ARTIFACT_ID);
+              assertThat(dependency.getVersion().getStringValue()).isEqualTo(CLOUD_LIB_VERSION);
+            });
+  }
+
+  @Test
+  public void addLibraries_withBom_writesDependencyWithNoVersion() {
+    ApplicationManager.getApplication()
+        .invokeAndWait(
+            () -> {
+              String bomVersion = "1.0.0-beta";
+              MavenDomProjectModel model = getMavenDomProjectModel(bomVersion);
+
+              List<MavenDomDependency> cloudLibraryDependencies =
+                  model.getDependencies().getDependencies();
+              assertThat(cloudLibraryDependencies.size()).isEqualTo(1);
+
+              MavenDomDependency dependency = cloudLibraryDependencies.get(0);
+              assertThat(dependency.getGroupId().getStringValue()).isEqualTo(CLOUD_LIB_GROUP_ID);
+              assertThat(dependency.getArtifactId().getStringValue())
+                  .isEqualTo(CLOUD_LIB_ARTIFACT_ID);
+              assertThat(dependency.getVersion().getStringValue()).isNull();
+            });
+  }
+
+  @Test
+  public void addLibraries_withNoBom_doesNotWriteBom() {
+    ApplicationManager.getApplication()
+        .invokeAndWait(
+            () -> {
+              MavenDomProjectModel model = getMavenDomProjectModel(null /*bomVersion*/);
+
+              assertThat(model.getDependencyManagement().getDependencies().getDependencies())
+                  .isEmpty();
+            });
+  }
+
+  @Test
+  public void addLibraries_withBom_writesBom() {
+    ApplicationManager.getApplication()
+        .invokeAndWait(
+            () -> {
+              String bomVersion = "1.0.0-beta";
+              MavenDomProjectModel model = getMavenDomProjectModel(bomVersion);
+
+              List<MavenDomDependency> dependencyManagementDependencies =
+                  model.getDependencyManagement().getDependencies().getDependencies();
+              assertThat(dependencyManagementDependencies.size()).isEqualTo(1);
+
+              MavenDomDependency bomDependency = dependencyManagementDependencies.get(0);
+              assertThat(bomDependency.getGroupId().getStringValue()).isEqualTo("com.google.cloud");
+              assertThat(bomDependency.getArtifactId().getStringValue())
+                  .isEqualTo("google-cloud-bom");
+              assertThat(bomDependency.getScope().getStringValue()).isEqualTo("import");
+              assertThat(bomDependency.getType().getStringValue()).isEqualTo("pom");
+              assertThat(bomDependency.getVersion().getStringValue()).isEqualTo(bomVersion);
+            });
+  }
+
+  private MavenDomProjectModel getMavenDomProjectModel(@Nullable String bomVersion) {
```

Comment:
Reading tests it's kind of a bit hard to see that this method actually adds a library into Maven DOM. Maybe add an optional library to add so when test is set up, it's easy to see we added a library, or use a name that mentions cloud library?

Reply:
agreed. lemme think about what the best option is

Reply:
changed it `createMavenModelWithCloudLibrary`. Let me know if this doesn't help the situation.

Reply:
awesome, reads easy!